### PR TITLE
OEL-2846: Check emails sent in contact form tests

### PR DESCRIPTION
**Pull request by AaronGilMartinez**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use Drupal\symfony_mailer\EmailInterface;
+
+/**
+ * Implements hook_mailer_TYPE_PHASE().
+ */
+function oe_showcase_contact_forms_mailer_contact_build(EmailInterface $email) {
+  if (empty($email->getTo())) {
+    return;
+  }
+  // This function removes extra spaces in recipient adresses added by contact
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module]:
```suggestion
  // This function removes extra spaces in recipient addresses added by contact
```

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  // module in mails sent by symfony_mailer.
+  // @see oe_showcase/build/core/modules/contact/src/MailHandler.php
+  $adresses = [];
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module]:
```suggestion
  $addresses = [];
```

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Implements hook_mailer_TYPE_PHASE().
+ */
+function oe_showcase_contact_forms_mailer_contact_build(EmailInterface $email) {
+  if (empty($email->getTo())) {
+    return;
+  }
+  // This function removes extra spaces in recipient adresses added by contact
+  // module in mails sent by symfony_mailer.
+  // @see oe_showcase/build/core/modules/contact/src/MailHandler.php
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module]: oe_showcase/build/core/... is not good, please use a fully qualified class name and ::method name attached, and a short explanation that lets me know what exactly MailHandler does and maybe why to attach the space. And please also look if there is an issue to fix it, and attach it here.

About the explanation, I honestly dont see where in that class the issue is, I didnt look deep into it but I rather expect it to be where the recipient email is set.

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.post_update.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+/**
+ * @file
+ * OE Showcase List pages post updates.
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/oe_showcase_contact_forms.post_update.php]: a copy paste error "List pages"

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php [hunk 5]
```diff
@@ -98,6 +108,96 @@ public function testCreateContactForm() {
     $this->assertStringContainsString('Belgium', $confirm_message);
     $this->assertStringContainsString('Example subject', $confirm_message);
     $this->assertStringContainsString('Example Message text', $confirm_message);
+
+    // Check sent mail.
+    $email = $this->readMail();
+    $page_url = $this->buildUrl('/pages/example-contact-form-page', ['absolute' => TRUE]);
+    $user_url = $user->toUrl('canonical', ['absolute' => TRUE])->toString();
+    // Configured recipients + Topic address.
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: I dont see anything related to "Topic address."

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php [hunk 6]
```diff
@@ -4,9 +4,13 @@
 
 namespace Drupal\Tests\oe_showcase_contact_forms\ExistingSite;
 
+use Drupal\symfony_mailer\Email;
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: phpstorm is indicating to me that Symfony\Component\Mime\Email is provided

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php [hunk 7]
```diff
@@ -98,6 +108,96 @@ public function testCreateContactForm() {
...
+      'Example Message text 2',
+      'Spain',
+      '123123123',
+      'Llama',
+    ], $email);
+    // Then check the copy mail.
+    $email = $this->readMail();
+    $this->assertTo($user->getEmail());
+    $this->assertMailLabelsTexts($email);
+    $this->assertMailTexts([
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: I dont like this very much, both are texts in the email body, so should be tested together, you can pass the dynamic strings as an array to a single function that tests the content of  the email body.

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php [hunk 8]
```diff
@@ -98,6 +108,96 @@ public function testCreateContactForm() {
...
+   *
+   * @param string[] $expected_texts
+   *   A list of expected text content to search in the mail body.
+   * @param \Drupal\symfony_mailer\Email $email
+   *   The mail.
+   */
+  protected function assertMailTexts(array $expected_texts, Email $email): void {
+    $crawler = new Crawler($email->getHtmlBody());
+    foreach ($expected_texts as $expected) {
+      $this->assertStringContainsString($expected, $crawler->text());
```
Comment (drishu) [inline modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: assertStringContainsString easily give false positives, so it is important what you pass to it, I see we getHtmlBody, we could add the wrapping tags, even better would be to define a mail template in a string and use sprintf to add in parameters. Lets discuss.

**Review by drishu — DISMISSED** [review]

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module (hunk 1 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module]: Changed

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module (hunk 2 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module]: Changed

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.post_update.php (hunk 4 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/oe_showcase_contact_forms.post_update.php]: Fixed

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php (hunk 6 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: Same than https://github.com/openeuropa/oe_subscriptions/pull/31#discussion_r1568365472

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php (hunk 5 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: Added

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php (hunk 7 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: Moved to the assertMailTexts()

File: modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php (hunk 8 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/tests/src/ExistingSite/ContactFormTest.php]: Added assertion with template with placeholders.

File: modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module (hunk 3 above)
Comment (AaronGilMartinez) [reply modules/oe_showcase_contact_forms/oe_showcase_contact_forms.module]: Changed explanation and added fully cualified names

**Review by brummbar — APPROVED** [review]

**Comment by brummbar** [comment]
We will have two followups in order to be able to turn on the Symfony Mailer override for contact forms.

**Pull request merged**